### PR TITLE
`Development`: Use directive for programming test status details and diff report

### PR DESCRIPTION
**Pull request by florian-glombik**
#### General
- [x] I have tested the changes on a test server
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context
Follow up on [Development: Add directive to fix types in exercise detail overview list #8644](https://github.com/ls1intum/Artemis/pull/8644)

### Description
* Added a component for ProgrammingDiffReportDetails
* Added a component for ProgrammingTestStatusDetails
* Using ExerciseDetails directive for ProgrammingDiffReportDetails
* Using ExerciseDetails directive for ProgrammingTestStatusDetails
* Added unit tests
* Added folder structure to detail components
* Made the ArtemisTranslate Pipe a standalone component

### Steps for Testing
Prerequisites:
- 1 Instructor

1. Navigate to the details page of a programming exercise
2. Verify that the template / solution result is displayed properly and triggering the build works as expected
3. Verify the template solution comparison is displayed properly

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots
The UI should not have changed

Screenshot 2024-08-30 at 16 53 55

## Summary by CodeRabbit

- **New Features**
	- Introduced two new components: ProgrammingDiffReportDetailComponent and ProgrammingTestStatusDetailComponent for enhanced display of programming exercise details.
	- Added translation support for short descriptions in the diff report component.
	- Implemented a button to view Git diff reports, conditionally displayed based on changes.
  
- **Bug Fixes**
	- Removed unused imports and functionalities related to Git diff reporting to streamline the interface.

- **Documentation**
	- Updated visibility of interfaces for better accessibility across modules.

- **Chores**
	- Restructured import paths for better organization and maintainability.
	- Enhanced test coverage for the ProgrammingDiffReportDetailComponent and ProgrammingTestStatusDetailComponent with unit tests.

### CODE REVIEW

File: src/main/webapp/app/detail-overview-list/detail-overview-list.component.ts [hunk 1]
```diff
@@ -92,15 +83,6 @@ export class DetailOverviewListComponent implements OnInit, OnDestroy {
         }, {});
     }
 
-    showGitDiff(gitDiff?: ProgrammingExerciseGitDiffReport) {
```
Comment (pzdr7) [inline src/main/webapp/app/detail-overview-list/detail-overview-list.component.ts:95]: Can you update the tests so they no longer try to call this method on the `DetailOverviewListComponent`? I think the build is failing

**Review by pzdr7 — CHANGES_REQUESTED** [review]

**Review by JohannesWt — APPROVED** [review]
Code 👍

**Review by az108 — APPROVED** [review]
Tested on TS2 works as expected 👍 
image

**Review by az108 — APPROVED** [review]
reapprove after merge

**Review by JohannesWt — APPROVED** [review]
Code reapprove

**Review by pzdr7 — APPROVED** [review]
Code

**Review by SimonEntholzer — APPROVED** [review]
Works on TS3 :+1:

**Pull request merged**